### PR TITLE
Ae/improve docs

### PR DESCRIPTION
**Pull request by alanoe**
Improve cartesian configuration documentation

### CODE REVIEW

File: docs/source/CartesianConfig.rst [hunk 1]
```diff
@@ -353,35 +353,30 @@ relevant, and will carry through any key/value substitutions
...
-In general, each individual combination of the defined variants provides
-the parameters for a single test. Testing proceeds in order, through
-each result, passing the set of keys and values through to the harness
-and test code. When examining Cartesian configuration files, it’s
-helpful to consider the earliest key definitions as “defaults”, then
-look to the end of the file for other top-level override to those
-values. If in doubt of where to define or set a key, placing it at the
-top indentation level, at the end of the file, will guarantee it is
-used.
+.. developing_configurations:
```
Comment (olavphibm) [inline docs/source/CartesianConfig.rst]: It looks like this should begin with an underscore.

File: docs/source/CartesianConfig.rst [hunk 2]
```diff
@@ -42,7 +42,7 @@ Variants
 ========
 
 A ‘variants’ stanza is opened by a ‘variants:’ statement. The contents
-of the stanza must be indented further left than the ‘variants:’
+of the stanza must be indented further right than the ‘variants:’
 statement. Each variant stanza or block defines a single dimension of
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: OK

File: docs/source/CartesianConfig.rst [hunk 3]
```diff
@@ -67,7 +67,7 @@ outside arrays.  For example::
 
 While combining, the parser forms names for each outcome based on
 prepending each variant onto a list. In other words, the first variant
-name parsed will appear as the left most name component. These names can
+name parsed will appear as the right most name component. These names can
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: OK

File: docs/source/CartesianConfig.rst [hunk 4]
```diff
@@ -41,11 +41,11 @@ definitions.
...
 
-A ‘variants’ stanza is opened by a ‘variants:’ statement. The contents
-of the stanza must be indented further right than the ‘variants:’
-statement. Each variant stanza or block defines a single dimension of
+A ‘variants’ block is opened by a ‘variants:’ statement. The contents
+of the block must be indented further right than the ‘variants:’
+statement. Each variant block defines a single dimension of
 the output array. When a Cartesian configuration file contains
-two variants stanzas, the output will be all possible combination's of
+two variants blocks, the output will be all possible combination's of
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: OK

File: docs/source/CartesianConfig.rst [hunk 5]
```diff
@@ -4,11 +4,16 @@
...
+In software testing, key/value pairs are usually used as test input parameters.
+
+Cartesian product is a mathematical operation which creates a set from
+multiple sets. Given sets A and B, the cartesian product of both is
+the set of all ordered pairs (a,b) where a belongs to A and b belongs to B.
+
+In Avocado VT, cartesian configuration is a method of defining key/value pairs
+which are later combined using the cartesian product concept. Each element of
+this product is used as input by a different test execution. The product can
+be filtered and adjusted prior to testing, with filters, dependencies, and
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: Bravo, excellent.

File: docs/source/CartesianConfig.rst [hunk 6]
```diff
@@ -87,10 +93,21 @@ combinations and names::
...
+Dict 7 elements::
+
+    key1 = foo
+    key2 = bar
+
+These combinations names can become quite long and, due to this, a 'short name'
+is also used. A combination short name represent the <TESTNAME> value used
+when a combination is recorded (see section Job Names and Tags. For convenience,
+variants which name begins with a ‘``@``’ do not prepend their name to
+'short name', only 'name'. This allows creating ‘shortcuts’ for
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: Again, excellent.

File: docs/source/CartesianConfig.rst [hunk 7]
```diff
@@ -427,6 +319,171 @@ values. If in doubt of where to define or set a key, placing it at the
...
+        dep = []
+        disk_interface = hda
+        guest_os = ubuntu
+        name = (disk_interface=hda).(guest_os=ubuntu)
+        shortname = (disk_interface=hda).(guest_os=ubuntu)
+
+.. _key_sub_arrays:
+
+Implicit keys
+-------------
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: Nice that you took the time to explain this.

File: docs/source/CartesianConfig.rst [hunk 8]
```diff
@@ -193,14 +196,75 @@ For example:
...
+      initrd = c
+    dict 2: Linux.OpenSuse
+      initrd = initrd_value
+
+Thus, this sets the ‘initrd’ key to ‘initrd_value’ in all combinations containing
+‘Linux’ followed (immediately or not) by ‘OpenSuse’.
+
+However, note that any of these filters may be used within variants as well.
+In this case, they are only evaluated when that variant name is
+selected for inclusion (implicitly or explicitly) by a higher-order.
```
Comment (lmr) [inline docs/source/CartesianConfig.rst]: Looks good.

**Comment by clebergnu** [comment]
Closing in favor of #979

**Pull request closed without merging**